test/worker_test.py

```diff
@@ -1296,3 +1298,358 @@ def store_task(t, error_msg):
         w._handle_next_task()
 
         self.assertEqual(result, [task])
+
+
+class PerTaskRetryPolicyBehaviorTest(unittest.TestCase):
+    def setUp(self):
+        self.per_task_retry_count = 2
+        self.default_retry_count = 1
+        self.sch = Scheduler(retry_delay=0.1, retry_count=self.default_retry_count, prune_on_get_work=True, record_task_history=False)
+
+        self.test_instance_id = str(uuid4())
+
+    def test_with_all_disabled_with_single_worker(self):
+        """
+            With this test, a case which has a task (A), requires two another tasks (B,C) which both is failed, is tested.
+
+            Task C has default retry_count which is 1, but Task B has retry_count at task level as 2.
+
+            This test is running on single worker
+        """
+        class C(luigi.Task):
+            task_namespace = self.test_instance_id
+
+            retry_index = 0
+
+            def run(self):
```

Comment:
Can you make your test cases shorter by having a task you inherit from. I've seen the `self.retry_index += 1` repeated many many times in this patch :)